### ISSUE TITLE
IntelliJ - Feathers source paths not added to Sourcepath for SDK

### ISSUE THREAD

**Issue by teotigraphix** [BowlerHatLLC/feathers-sdk#7]
If you look at Project Structure -> FeathersSDK -> Sourcepath, most sources are listed including starling but feathers and feathers-mxml are not there.

Would be nice to have those automatically added since middle mouse button is majic in IJ for looking at definitions.

**Comment by joshtynjala** [comment]
I think it's because I use "source" instead of "src" for the directory name.

I could possibly change the directory name in the mxml branch of Feathers, but I wonder if that will cause merge issues in the future.

**Comment by joshtynjala** [comment]
Since the [Flex plugin for IntelliJ IDEA](https://github.com/JetBrains/intellij-plugins/tree/master/flex) is open source, it may be possible to contribute a fix for this issue.

I only took a quick look, but it appears that [`createSourceRoot()` method of the `FlexModuleBuilder` class](https://github.com/JetBrains/intellij-plugins/blob/master/flex/src/com/intellij/lang/javascript/flex/FlexModuleBuilder.java#L254) may be where IDEA tries to find the SDKs source roots. This method searches for a path with "src". If this directory doesn't exist, it may be enough to search for "source" too.

**Comment by teotigraphix** [comment]
Kewl, yeah man my schedule is tight until January basically(Using Feathers SDK for Caustic apps).

I have created/helped create a Plugin in IDEA before with Fred(Randori as/js using FalconJX), so I know my way around IJ code but it's the time factor right now and to do a pull request you have to have the whole shabang setup and working for Jetbrains to accept it.

**Comment by joshtynjala** [comment]
I totally understand. I just wanted to record the results of a few minutes of investigation, since it might help someone get started.

**Comment by teotigraphix** [comment]
Yeah I was just sayin. :) I would love this coming year to help get Feathers SDK a first class citizen in IJ and see if we could tweak design view and get it working, don't know how that would go but I bet we could do it. :) Once I get off the ground with these apps I will have a lot more lab time to extend what I have already done. Besides contributing to Feathers once my company is rolling again will only be an investment for my future.
